core/src/main/java/com/graphhopper/routing/util/FerrySpeedCalculator.java

```diff
@@ -7,66 +7,55 @@
-                // specify the duration in minutes, but actually using months
-                if (calculatedTripSpeed > 0.01d) {
-                    if (calculatedTripSpeed > maxSpeed) {
-                        return maxSpeed;
-                    }
-                    // If the speed is lower than the speed we can store, we have to set it to the minSpeed, but > 0
-                    if (Math.round(calculatedTripSpeed) < speedFactor / 2) {
-                        return speedFactor / 2;
-                    }
-
-                    return Math.round(calculatedTripSpeed);
-                } else {
-                    long lastId = way.getNodes().isEmpty() ? -1 : way.getNodes().get(way.getNodes().size() - 1);
-                    long firstId = way.getNodes().isEmpty() ? -1 : way.getNodes().get(0);
-                    if (firstId != lastId)
-                        LOGGER.warn("Unrealistic long duration ignored in way with way ID=" + way.getId() + " : Duration tag value="
-                                + way.getTag("duration") + " (=" + Math.round(duration / 60d) + " minutes)");
-                    durationInHours = 0;
-                }
+            // If duration is available we can calculate the speed. We make it slower by a factor of 1.4 to account
+            // for waiting time.
+            double calculatedTripSpeed = distanceInKm / durationInHours / 1.4;
+            // Plausibility check especially for the case of wrongly used PxM format with the intention to
+            // specify the duration in minutes, but actually using months
+            if (calculatedTripSpeed > 0.01d) {
+                if (calculatedTripSpeed > maxSpeed)
+                    return maxSpeed;
+                // If the speed is lower than the speed we can store, we have to set it to the minSpeed, but > 0
+                if (Math.round(calculatedTripSpeed) < minSpeed)
+                    return minSpeed;
+                return Math.round(calculatedTripSpeed);
+            } else {
+                LOGGER.warn("Unrealistic long duration ignored in way with way ID=" + way.getId() + " : Duration tag value="
+                        + way.getTag("duration") + " (=" + Math.round(duration / 60d) + " minutes)");
             }
         }
 
-        if (durationInHours == 0) {
-            if (estimatedLength != null && estimatedLength.doubleValue() <= 300)
-                return speedFactor / 2;
+        // duration was not present or calculated trip speed was too small
+        if (distanceInKm <= 0.3)
+            return minSpeed;
+        else
             // unknown speed -> put penalty on ferry transport
             return unknownSpeed;
-        } else if (durationInHours > 1) {
-            // lengthy ferries should be faster than short trip ferry
-            return longSpeed;
-        } else {
-            return shortSpeed;
-        }
```

Comment:
> lengthy ferries should be faster than short trip ferry

I still think this is something we might have to do. If the duration tag is missing we probably *should* apply a faster speed for long ferries than for short ones. Especially because the `unknownSpeed` is very slow (5km/h). There is probably no ferry that crosses the sea at 5km/h. But so far we only applied this long/shortSpeed when the distance was missing (which never happened anyway), *not* when the duration was missing, or at least we only distinguished between "very slow/`unknownSpeed` and "as slow as possible"/`speedFactor / 2` in this case. 

We probably need to figure out how often the duration is missing and if it is sometimes missing also for longer ferries. I can imagine that mappers skip the duration tag for short ferries that just cross a river, but for longer distance ferries this seems like a mapping error and we could even print a warning, report this to OSM and/or skip the ferry in this case. I opened #2532 for this.

Reply:
> We probably need to figure out how often the duration is missing and if yes if it is sometimes missing also for longer ferries.

It is in fact missing in very many cases and also for longer ferries like this one: https://www.openstreetmap.org/way/999670971. Simply falling back to the `unknownSpeed=5km/h` is thus not a good idea. Here is the [route](https://graphhopper.com/maps/?point=35.78969%2C-5.804558&point=36.352186%2C-3.393402&point=36.835668%2C-1.538086&point=37.688167%2C0.626221&point=38.225235%2C1.966553&point=38.766933%2C3.378296&point=42.747012%2C7.393799&point=43.411033%2C7.959595&point=43.886016%2C8.363342&point=44.262558%2C8.449516&locale=en-US&elevation=true&profile=car&use_miles=false&selected_detail=Elevation&layer=OpenStreetMap). The average speed is 6km/h (unknownSpeed=5, but rounded to 6 because of speedFactor=2). It takes more than eleven days when really this ferry only needs around two (35km/h).